src/styles/blocks.scss

```diff
@@ -0,0 +1,64 @@
+* {
+  box-sizing: border-box;
+}
+
+@each $name, $index in $dayOfWeek {
+  .calendar__start-day--#{$name}
+    .calendar__day:first-child {
+    margin-left: calc((#{$squareBox} + #{$gap}) * (#{$index} - 1));
+    }
+}
+
+@for $days from 28 through $monthLength {
+  .calendar__month-length--#{$days}
+    .calendar__day:nth-child(n + #{$days + 1}) {
+    display: none;
+    }
+}
+
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+}
+
+.calendar {
+  width: calc(
+    (#{$squareBox} * #{$dayInWeek})
+    + ((#{$dayInWeek} * #{$gap}) - #{$gap})
+    + (#{$paddingCalendar * 2})
+  );
+  padding: $paddingCalendar;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+
+  &__day {
+    width: $squareBox;
+    height: $squareBox;
+    border: $borderWidth solid #000;
+    background-color: #eee;
+    transition: transform 0.5s, background-color 0.5s;
+
+    @for $i from 1 through $monthLength {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+      }
+    }
+
+    &::before {
+      display: flex;
+      justify-content: center;
+      line-height: calc(#{$squareBox} - (#{$borderWidth} * 2));
+
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
```

Comment:
It is better to use one format for specifying colors, and since you have already started with hex format, continue like that